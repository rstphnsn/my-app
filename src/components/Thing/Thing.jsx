import React from 'react'

const Thing = ({ lazy }) => (
  <div>
    Some {lazy ? 'lazy' : ''} thing
  </div>
)

export default Thing