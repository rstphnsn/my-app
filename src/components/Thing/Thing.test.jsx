import React from 'react';
import { shallow } from 'enzyme';

import Thing from '.'

describe('Thing component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Thing />);
  });

  it('renders without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  describe('when `lazy` prop is truthy', () => {
    it('renders without crashing', () => {
      component = shallow(<Thing lazy />);
      expect(component).toMatchSnapshot();
    });
  });
});
