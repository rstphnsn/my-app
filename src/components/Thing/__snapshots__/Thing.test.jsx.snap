// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thing component renders without crashing 1`] = `
<div>
  Some 
   thing
</div>
`;

exports[`Thing component when \`lazy\` prop is truthy renders without crashing 1`] = `
<div>
  Some 
  lazy
   thing
</div>
`;
